### PR TITLE
2fa redirection

### PR DESCRIPTION
**Pull request by eprei**
Implementation of routing based on actual
login status.

Being on the SignIn page, if the user
accesses successfully with login 42 and
does not have 2fa enabled, he is
redirected to the profile page.

Being on the SignIn page, if the user
successfully logs in with login 42 and has
2fa enabled, he/she will be redirected to
the 2fa verification page. If he/she
enters a correct code, he/she is
redirected to the profile page.

If the user is already fully authenticated
but tries to log in to the SignIn page,
he/she is redirected to the profile page.

Added logout button (need to improve the
css because it is not readable when
hovering the mouse over the button).This
button is available in the Navbar and in
the 2fa verification page.

### CODE REVIEW

File: frontend/src/components/navigation/LogOutButton.module.css [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+.sign svg {
+    width: 17px;
+}
+
+.sign svg path {
+    fill: white;
+}
+/* text */
+.text {
+    position: absolute;
```
Comment (fire-poy) [inline frontend/src/components/navigation/LogOutButton.module.css:35]: position: relative; or position: right;  works well (the text don't hide)
But the effecs is a little annoying to look, but I don't know how to make it better

File: frontend/src/components/navigation/LogOutButton.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    return (
+        <button
+            className={styles.Btn}
+            onClick={handleLogout}
+            disabled={isLoading}
+        >
+            <div className={styles.sign}>
+                <svg viewBox="0 0 512 512">
```
Comment (fire-poy) [inline frontend/src/components/navigation/LogOutButton.tsx:39]: I think you can add viewBox to assets

**Review by fire-poy — APPROVED** [review]
good work!!

File: frontend/src/components/navigation/LogOutButton.module.css (hunk 1 above)
Comment (eprei) [reply frontend/src/components/navigation/LogOutButton.module.css:35]: For me, 'position: right' seems to work fine. Thanks for the suggestion!

File: frontend/src/components/navigation/LogOutButton.tsx (hunk 2 above)
Comment (eprei) [reply frontend/src/components/navigation/LogOutButton.tsx:39]: I'm not sure what that means

**Review by t-h2o — COMMENTED** [review]
:broom: :question:

File: frontend/src/App.tsx [hunk 3]
```diff
@@ -14,14 +14,23 @@ import { useAppDispatch } from './store/types'
...
+import ProtectedSignIn from './ProtectedSignIn'
 
 const router = createBrowserRouter([
     {
         path: '/',
         element: <RootLayout />,
         errorElement: <ErrorPage />,
         children: [
-            { index: true, element: <SignIn /> },
+            // { index: true, element: <SignIn /> },
```
[the remark this answers is not in the document]
Comment (eprei) [reply frontend/src/App.tsx:25]: Of course

**Review by RobinBurri — APPROVED** [review]
👍

**Pull request merged**